### PR TITLE
Scan code rework

### PR DESCRIPTION
**Pull request by DAloG**
Move all AV related stuff into session object. Connect session - vc - vm layers.

### CODE REVIEW

**Comment by DAloG** [comment]
@xNekOIx What do you think about these changes?

File: iOS/UZTickets/src/scan/UZTScanSession.m [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    NSParameterAssert(callback != nil);
+    
+    [self.session startRunning];
+    self.currentCallback = callback;
+}
+
+- (void)stopScanning
+{
+    NSAssert(self.session.isRunning, @"Session not started yet");
+    self.currentCallback = nil;
```
Comment (xNekOIx) [inline iOS/UZTickets/src/scan/UZTScanSession.m]: didn't you miss ```[self.session stopRunning]```?

File: iOS/UZTickets/src/scan/UZTScanSession.m [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        NSError* error = nil;
+        self.deviceInput = [AVCaptureDeviceInput deviceInputWithDevice:self.device
+                                                                 error:&error];
+        NSAssert(error == nil, @"Error during input inititialization");
+        NSAssert(self.deviceInput != nil, @"Mising device inout");
+    }
+
+    self.metadataOutput = [AVCaptureMetadataOutput new];
+    self.session = [AVCaptureSession new];
+    self.previewLayer = [AVCaptureVideoPreviewLayer layerWithSession:self.session];
```
Comment (xNekOIx) [inline iOS/UZTickets/src/scan/UZTScanSession.m]: is it possible to run session and code recognition without ```AVCaptureVideoPreviewLayer```?

File: iOS/UZTickets/src/scan/UZTScanViewController.m [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@end
+
+@implementation UZTScanViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.scanningSession = [UZTScanSession new];
+    [self.view.layer insertSublayer:self.scanningSession.layer atIndex:0];
```
Comment (xNekOIx) [inline iOS/UZTickets/src/scan/UZTScanViewController.m]: I still recon that managing layers separately from view is more OS X way of working with display tree.

**Comment by xNekOIx** [comment]
@DAloG  Why did you make it in separate branch and pull request?

**Pull request merged**